Detect devpod cli on windows

Follow-up issue on the discussion [here](https://procial.tchncs.de/notes/9y4507rewfmb0fu6).

Searching for the bin is happening [here](https://github.com/3timeslazy/vscodium-devpodcontainers/blob/main/src/devpod/bin.ts), correct?

I have some time to have a look on a windows machine. Can you give me a starter on how to debug an extension? I tried running the extension from within the repository without any success. Something about missing a module:
```
Activating extension '3timeslazy.vscodium-devpodcontainers' failed: Cannot find module 'c:\Users\dev\Documents\vscodium-devpodcontainers\dist\extension.js' Require stack: - c:\Users\dev\scoop\apps\vscodium\1.93.1.24256\resources\app\out\bootstrap-fork.js.
```
Any tips/idea on what Im doing wrong here?

--- a/package.json
+++ b/package.json
@@ -99,6 +99,7 @@
     "@types/mocha": "^10.0.7",
     "@types/node": "^20.16.5",
     "@types/vscode": "^1.80.0",
+    "@types/which": "^3.0.4",
     "@typescript-eslint/eslint-plugin": "^7.14.1",
     "@typescript-eslint/parser": "^7.11.0",
     "@vscode/test-cli": "^0.0.9",

--- a/package.json
+++ b/package.json
@@ -95,8 +95,9 @@
     "test": "vscode-test"
   },
   "devDependencies": {
+    "@types/command-exists": "^1.2.3",
     "@types/mocha": "^10.0.7",
-    "@types/node": "20.x",
+    "@types/node": "^20.16.5",
     "@types/vscode": "^1.80.0",
     "@typescript-eslint/eslint-plugin": "^7.14.1",
     "@typescript-eslint/parser": "^7.11.0",
@@ -105,10 +106,11 @@
     "esbuild": "^0.21.5",
     "eslint": "^8.57.0",
     "npm-run-all": "^4.1.5",
-    "typescript": "^5.4.5"
+    "typescript": "^5.6.2"
   },
   "dependencies": {
     "@vscode/vsce": "^2.29.0",
+    "command-exists": "^1.2.9",
     "jsonc-parser": "^3.3.1"
   }
 }

--- a/src/devpod/bin.ts
+++ b/src/devpod/bin.ts
@@ -1,7 +1,7 @@
 import vscode from 'vscode';
 import path from 'path';
 import fs from 'fs';
-import commandExists from 'command-exists';
+import which from 'which';
 
 export async function installDevpod() {
 	const install = { title: 'Install' };
@@ -30,8 +30,6 @@ export async function installDevpod() {
 	}
 }
 
-export function devpodBinExists(): Promise<boolean> {
-	return commandExists('devpod')
-    .then(() => true)
-    .catch(() => false);
+export function devpodBinExists(): boolean {
+	return which.sync('devpod', { nothrow: true }) !== null;
 }

--- a/src/devpod/bin.ts
+++ b/src/devpod/bin.ts
@@ -1,6 +1,7 @@
 import vscode from 'vscode';
 import path from 'path';
 import fs from 'fs';
+import commandExists from 'command-exists';
 
 export async function installDevpod() {
 	const install = { title: 'Install' };
@@ -29,28 +30,8 @@ export async function installDevpod() {
 	}
 }
 
-export function devpodBinExists(): boolean {
-	const envPath = process.env['PATH'] || '';
-	const paths = envPath?.split(path.delimiter);
-	let s = '';
-	for (const p of paths) {
-		const binpath = path.join(p, 'devpod');
-		if (executableFileExists(binpath)) {
-			return true;
-		}
-	}
-	return false;
-}
-
-function executableFileExists(filePath: string): boolean {
-	let exists = true;
-	try {
-		exists = fs.statSync(filePath).isFile();
-		if (exists) {
-			fs.accessSync(filePath, fs.constants.F_OK | fs.constants.X_OK);
-		}
-	} catch (e) {
-		exists = false;
-	}
-	return exists;
+export function devpodBinExists(): Promise<boolean> {
+	return commandExists('devpod')
+    .then(() => true)
+    .catch(() => false);
 }

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -110,20 +110,20 @@ async function openContainer(recreate: boolean = false) {
 
 	await upDevpod({
 		configPath: pick.path.replace(workspace.uri.path, ''),
-		workspaceFolder: workspace.uri.path,
+		workspaceFolder: workspace.uri.fsPath,
 		recreate: recreate,
 	});
 
 	const devpods = await listDevpods();
 	// TODO: more then one devpod in the directory.
-	const devpod = devpods.find((d) => d.source.localFolder === workspace.uri.path);
+	const devpod = devpods.find((d) => d.source.localFolder === workspace.uri.fsPath);
 	if (!devpod) {
 		vscode.window.showErrorMessage('Unknown error: no devpod found');
 		return;
 	}
 	const devpodHost = `${devpod.id}.devpod`;
 
-	const customizations = parseCustomizations(pick.path);
+	const customizations = parseCustomizations(pick.fsPath);
 	const exts = customizations.extensions;
 	const installExtArgs = [];
 	const registryExts = [];

